src/host/rpc_connections.h

```diff
@@ -46,50 +138,89 @@ namespace asynchost
           serializer::ByteRange{data, len});
       }
 
-      void on_disconnect()
+      void on_disconnect() override
       {
         LOG_DEBUG_FMT("rpc disconnect {}", id);
         cleanup();
       }
 
       void cleanup()
       {
-        RINGBUFFER_WRITE_MESSAGE(tls::tls_close, parent.to_enclave, id);
+        if (getConnTypeID<ConnType>() == CONN_TCP)
+        {
+          RINGBUFFER_WRITE_MESSAGE(tls::tls_close, parent.to_enclave, id);
+        }
       }
     };
 
-    class RPCServerBehaviour : public TCPServerBehaviour
+    class RPCServerBehaviour : public SocketBehaviour<ConnType>
     {
     public:
       RPCConnections& parent;
-      int64_t id;
+      ConnID id;
 
-      RPCServerBehaviour(RPCConnections& parent, int64_t id) :
+      RPCServerBehaviour(RPCConnections& parent, ConnID id) :
+        SocketBehaviour<ConnType>("RPC Client", getConnTypeName<ConnType>()),
         parent(parent),
         id(id)
       {}
 
-      void on_listening(
-        const std::string& host, const std::string& service) override
+      void on_accept(ConnType& peer) override
       {
-        LOG_INFO_FMT("Listening for RPCs on {}:{}", host, service);
-      }
+        // UDP connections don't register peers
+        if (getConnTypeID<ConnType>() == CONN_UDP)
```

Comment:
If you make `getConnTypeID` constexpr, this can be `if constexpr (...)`, and perhaps even `if constexpr (std::is_same_v<ConnType, asynchost::TCP>())` (if the `TCP` and `UDP` types are safely visible here, and the enum's not used for anything other than this type of switch).

Reply:
Done.